web/react/components/create_post.jsx

```diff
@@ -32,7 +32,7 @@ module.exports = React.createClass({
         post.message = this.state.messageText;
 
         // if this is a reply, trim off any carets from the beginning of a message
-        if (this.state.rootId && post.message.startsWith("^")) {
+        if (this.state.rootId && post.message.lastIndexOf("^", 0) === 0) {
```

Comment:
Should this be indexOf and not lastIndexOf? This will be false if the message starts with `^^` or if the message starts with `^` and has another `^` somewhere in it.